pandas/io/parquet.py

```diff
@@ -107,6 +102,11 @@ def write(
         # write_to_dataset does not support a file-like object when
         # a directory path is used, so just pass the path string.
         if partition_cols is not None:
+            if is_fsspec_url(path) and "filesystem" not in kwargs:
```

Comment:
Can you leave a comment explaining this `"filesystem" not in kwargs` check? It's not obvious to me why it's needed.

Reply:
In `fsspec`, you can specify the exact protocol you would like beyond that inferred from the URL. Given that we don't pass storage_options through yet, perhaps this gives more flexibility than required and I can remove it.

Reply:
Sorry, edit on that: this is the filesystem parameter (i.e., an actual instance) to pyarrow. I have no idea if people might currently be using that.

Reply:
Ah, you're saying the user could pass a filesystem like

```
df.to_parquet(..., filesystem=filesystem)
```

That certainly seems possible. Could you ensure that we have a test for that?